src/metabase/models/serialization.clj

```diff
@@ -676,24 +687,37 @@
   (fn [ingested _]
     (ingested-model ingested)))
 
-(defn- load-by-spec [ingested]
-  (let [model-name (ingested-model ingested)
-        spec       (make-spec model-name)]
-    (when spec
-      (into (select-keys ingested (:copy spec))
-            (for [[k [_ser des]] (:transform spec)]
-              [k (des (get ingested k))])))))
+(defn- xform-by-spec [model-name ingested]
+  (binding [*current* ingested]
+    (let [spec (make-spec model-name nil)]
+      (when spec
+        (-> (select-keys ingested (:copy spec))
+            (into (for [[k [_ser des :as transform]] (:transform spec)
+                        :when (not (::post (meta transform)))
```

Comment:
The use of `::post` seems restricted to handling nested entities. I wonder if you've thought of other use cases for it? As far as I can think of, there's no reason to do post-processing unless it's a nested entity. So maybe we can be more specific here and use a more specific marker that signals a transform is for a nested entity. 

Again, one solution is to mark nested entities as distinct from other transform keys in the spec, like this:

```
(defmethod serdes/make-spec "Dashboard" [_model-name opts]
  {:copy      [:archived :archived_directly :auto_apply_filters :cache_ttl :caveats :collection_position :created_at
               :description :embedding_params :enable_embedding :entity_id :initially_published_at :name
               :points_of_interest :position :public_uuid :show_in_getting_started :width]
   :skip      [;; those stats are inherently local state
               :view_count :last_viewed_at]
   :transform {:collection_id     (serdes/fk :model/Collection)
               :creator_id        (serdes/fk :model/User)
               :made_public_by_id (serdes/fk :model/User)
               :parameters        [serdes/export-parameters serdes/import-parameters]}
   :nested    {:dashcards {:model :model/DashboardCard,
                           :fk    :dashboard_id}
               :tabs      {:model :model/DashboardTab
                           :fk    :dashboard_id}}})
```